engine/access/rpc/backend/backend_accounts.go

```diff
@@ -13,102 +15,149 @@ import (
 	"github.com/onflow/flow-go/engine/access/rpc/connection"
 	"github.com/onflow/flow-go/engine/common/rpc"
 	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	fvmerrors "github.com/onflow/flow-go/fvm/errors"
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/execution"
 	"github.com/onflow/flow-go/state/protocol"
 	"github.com/onflow/flow-go/storage"
 )
 
 type backendAccounts struct {
+	log               zerolog.Logger
 	state             protocol.State
 	headers           storage.Headers
 	executionReceipts storage.ExecutionReceipts
 	connFactory       connection.ConnectionFactory
-	log               zerolog.Logger
 	nodeCommunicator  Communicator
+	scriptExecutor    execution.ScriptExecutor
+	scriptExecMode    ScriptExecutionMode
 }
 
+// GetAccount returns the account details at the latest sealed block.
+// Alias for GetAccountAtLatestBlock
 func (b *backendAccounts) GetAccount(ctx context.Context, address flow.Address) (*flow.Account, error) {
 	return b.GetAccountAtLatestBlock(ctx, address)
 }
 
+// GetAccountAtLatestBlock returns the account details at the latest sealed block.
 func (b *backendAccounts) GetAccountAtLatestBlock(ctx context.Context, address flow.Address) (*flow.Account, error) {
-
-	// get the latest sealed header
-	latestHeader, err := b.state.Sealed().Head()
+	sealed, err := b.state.Sealed().Head()
```

Comment:
just want to double-check, this state used here is the same state as used by script executor right?

Reply:
I don't think the script executor uses `state` directly, just `headers`. but they are both populated by the follower engine